src/parser/xasl_generation.c

```diff
@@ -18213,6 +18229,7 @@ pt_to_insert_xasl (PARSER_CONTEXT * parser, PT_NODE * statement)
 	  (void) memcpy (xasl->class_oid_list + 1, aptr->class_oid_list, sizeof (OID) * aptr->n_oid_list);
 	  (void) memcpy (xasl->class_locks + 1, aptr->class_locks, sizeof (int) * aptr->n_oid_list);
 	  (void) memcpy (xasl->tcard_list + 1, aptr->tcard_list, sizeof (int) * aptr->n_oid_list);
+	  XASL_SET_FLAG (xasl, aptr->flag & XASL_INCLUDES_TDE_CLASS);
```

Comment:
xasl.flag = 1, aptr->flag = 0 then it is overwritten to 0. so is it needed to check XASL_IS_FLAGED(aptr->flag)?

Reply:
OMG, you're right. Thanks, I'll fix it.

Reply:
Ah, `XASL_SET_FLAG()` do `|=` to xasl->flags. If aptr->flag is 0, `aptr->flag` ends up still 1.

Reply:
OK